Libplanet/Net/Swarm.cs

```diff
@@ -363,62 +272,41 @@ private set
                     _replyQueue.Dispose();
                     _router.Dispose();
 
-                    foreach (DealerSocket s in _dealers.Values)
+                    foreach ((DateTimeOffset, DealerSocket) pair in _dealers.Values)
                     {
-                        s.Dispose();
+                        pair.Item2.Dispose();
                     }
 
                     _dealers.Clear();
 
                     Running = false;
                 }
+                else if (!(_protocol is null))
+                {
+                    // FIXME: Should be move to destructor
+                    _router.Dispose();
+                }
```

Comment:
How about calling `_router.Dispose()` outside of this if statement?

Reply:
It would not guarantee that this swarm is prepared.